lib/Backends/NNPI/InferencePool.cpp

```diff
@@ -73,7 +130,8 @@ InferenceThreadEnv::~InferenceThreadEnv() {
 void InferenceThreadEnv::execute(RunIdentifierTy runId,
```

Comment:
Not for now but for future releases -- the length of these functions in InferencePool are pretty high -- e.g. this one is 430 lines. I'd like for this to be refactored at some point -- we should try to carve out big chunks here to other functions, e.g. add a `prepareInputsForDevice()` and a `prepareInputsForICEREF()` so we can make the flow here clearer and less error prone.

Reply:
That's already part of an upcoming refactor - great minds... :)